### PR TITLE
Correlation metrics for activity on sparse GSC network

### PR DESCRIPTION
**Pull request by FishAres**

### CODE REVIEW

File: nupic/research/frameworks/pytorch/model_utils.py [hunk 1]
```diff
@@ -116,6 +121,13 @@ def train_model(
         else:
             loss.backward()
 
+        if len(freeze_params) > 0:
```
Comment (subutai) [inline nupic/research/frameworks/pytorch/model_utils.py:124]: should be `if freeze_params is not None:`

File: nupic/research/frameworks/continuous_learning/cl_utils.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+import sys
+import time
+
+import torch
+import torch.nn.functional as F
+from tqdm import tqdm
+
+
+def train_multi_model(
```
Comment (subutai) [inline nupic/research/frameworks/continuous_learning/cl_utils.py:30]: Do you need this function in cl_utils anymore?

File: nupic/research/frameworks/continuous_learning/cl_utils.py [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+                                num_images=num_images, time_string=time_string)
+        del loss
+        t0 = time.time()
+
+    if progress_bar is not None:
+        loader.n = loader.total
+        loader.close()
+
+
+def train_model(
```
Comment (subutai) [inline nupic/research/frameworks/continuous_learning/cl_utils.py:144]: Do you need this function in cl_utils anymore?

File: nupic/research/frameworks/pytorch/model_utils.py [hunk 4]
```diff
@@ -55,6 +56,10 @@ def train_model(
     :param batches_in_epoch: Max number of mini batches to train.
     :param device: device to use ('cpu' or 'cuda')
     :type device: :class:`torch.device
+    :param freeze_params: Parameters to freeze at specified indices
```
Comment (subutai) [inline nupic/research/frameworks/pytorch/model_utils.py:59]: `List of parameters to freeze at specified indices`

File: nupic/research/frameworks/pytorch/model_utils.py [hunk 5]
```diff
@@ -55,6 +56,10 @@ def train_model(
...
+     - freeze_params[0] -> network module
```
Comment (subutai) [inline nupic/research/frameworks/pytorch/model_utils.py:60]: `params[0]` and `params[1]` not `freeze_params`

File: nupic/research/frameworks/continuous_learning/cl_utils.py (hunk 2 above)
Comment (FishAres) [reply nupic/research/frameworks/continuous_learning/cl_utils.py:30]: i thought i deleted cl_utils.py, i'll fix that

File: nupic/research/frameworks/pytorch/model_utils.py (hunk 5 above)
Comment (FishAres) [reply nupic/research/frameworks/pytorch/model_utils.py]: merged your comments :)

**Pull request merged**